opencog/atoms/bind/PatternLink.cc

```diff
@@ -482,9 +475,58 @@ void PatternLink::unbundle_virtual(const std::set<Handle>& vars,
 {
 	for (const Handle& clause: clauses)
 	{
-		bool is_virtual = false;
 		bool is_black = false;
 
+		// Check if all links from top to bottom are either VirtualLink,
+		// logical links, EvaluationLink with GroundedPredicateNode, or
+		// ExecuationOutputLink; cause otherwise none of the evaluatable
+		// can be evaluated and must be matched directly.
+		std::function<bool (const Handle&)> is_virtualable = [&](const Handle& h)
```

Comment:
well, but this is what "trace_connectives" did; you re-wrote it, but made it less general ... 

Again, different logics, e.g. the kripke logics, or anything based on probability, are going to use different connectives, and we should allow these general cases to be handled.

A little bit off-topic, but treating PLN so that is was more kripke-like might make forward and backward chaining easier to implement: at each time-step, one has a different universe of things that can be "true" (follow from inference).